api/golang/engine/lib/kurtosis_context/kurtosis_context.go

```diff
@@ -346,12 +353,13 @@ func newEnclaveContextFromEnclaveInfo(
 	portalClient portal_api.KurtosisPortalClientClient,
 	enclaveInfo *kurtosis_engine_rpc_api_bindings.EnclaveInfo,
 ) (*enclaves.EnclaveContext, error) {
+	// for remote contexts, we need to tunnel the APIC port to the local machine
 	currentContext, err := store.GetContextsConfigStore().GetCurrentContext()
 	if err == nil {
 		// for remote contexts, we need to tunnel the APIC port to the local machine
 		if store.IsRemote(currentContext) && portalClient != nil {
 			apicGrpcPort := enclaveInfo.GetApiContainerHostMachineInfo().GetGrpcPortOnHostMachine()
-			forwardApicPortArgs := portal_constructors.NewForwardPortArgs(apicGrpcPort, apicGrpcPort, &apicPortTransportProtocol)
+			forwardApicPortArgs := portal_constructors.NewForwardPortArgs(apicGrpcPort, apicGrpcPort, ApicRemoteEndpointType, &apicPortTransportProtocol, &ForwardPortWaitUntilReady)
```

Comment:
Pass the remote endpoint type and forward port wait until ready flag. We want to wait until the port is forwarded since we are using the connection right after.